File: ios/Classes/Bridge/ChangeLanguageBridge.swift
import Usercentrics

struct ChangeLanguageBridge : MethodBridge {

    let name: String = "changeLanguage"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        guard let argument = call.arguments as? String else {
            result(FlutterError(code: "usercentrics_flutter_changeLanguage_error",
                                message: "Invalid Parameter",
                                details: nil ))
            return
        }

        usercentrics.shared.changeLanguage(language: argument) {
            result(nil)
        } onFailure: { error in
            result(FlutterError(code: "usercentrics_flutter_changeLanguage_error",
                                message: error.localizedDescription,
                                details: nil ))
        }
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/ChangeLanguageBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class ChangeLanguageBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    companion object {
        private const val changeLanguageErrorCode = "usercentrics_flutter_changeLanguage_error"
    }

    override val name: String
        get() = "changeLanguage"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.changeLanguage(
            language = call.arguments as String,
            onSuccess = {
                result.success(null)
            },
            onFailure = {
                result.error(changeLanguageErrorCode, it.message, it)
            }
        )
    }
}